pkg/scheduler/framework/interface_test.go

```diff
@@ -120,3 +123,82 @@ func TestPluginToStatusMerge(t *testing.T) {
 		}
 	}
 }
+
+func TestIsStatusEqual(t *testing.T) {
+	tests := []struct {
+		name string
+		x, y *Status
+		want bool
+	}{
+		{
+			name: "two nil should be equal",
+			x:    nil,
+			y:    nil,
+			want: true,
+		},
+		{
+			name: "nil should be equal to success status",
+			x:    nil,
+			y:    NewStatus(Success),
+			want: true,
+		},
+		{
+			name: "nil should not be equal with status except success",
+			x:    nil,
+			y:    NewStatus(Error, "internal error"),
+			want: false,
+		},
+		{
+			name: "one status should be equal to itself",
+			x:    errorStatus,
+			y:    errorStatus,
+			want: true,
+		},
+		{
+			name: "same type statuses without reasons should be equal",
+			x:    NewStatus(Success),
+			y:    NewStatus(Success),
+			want: true,
+		},
+		{
+			name: "statuses with same reasons should not be equal",
+			x:    NewStatus(Unschedulable, "unschedulable"),
+			y:    NewStatus(Unschedulable, "unschedulable"),
+			want: false,
+		},
+		{
+			name: "error statuses with same fields should not be equal",
+			x:    NewStatus(Error, "error"),
+			y:    NewStatus(Error, "error"),
+			want: false,
+		},
+		{
+			name: "statuses with different reasons should not be equal",
+			x:    NewStatus(Unschedulable, "unschedulable"),
+			y:    NewStatus(Unschedulable, "unschedulable", "injected filter status"),
+			want: false,
+		},
+		{
+			name: "statuses with different codes should not be equal",
+			x:    NewStatus(Error, "internal error"),
+			y:    NewStatus(Unschedulable, "internal error"),
+			want: false,
+		},
+		{
+			name: "wrap error status should be equal with original one",
+			x:    errorStatus,
+			y:    AsStatus(fmt.Errorf("error: %w", errorStatus.AsError())),
+			want: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := StatusEqual(tt.x, tt.y); got != tt.want {
```

Comment:
can we use `cmp.Equal` instead?

If you make the comparison a member function, `cmp` should be able to use it, I think.

Reply:
I prefer to test  equality function `StatusEqual` directly here to make it scalable as a unit test.

Reply:
the whole point of adding a Status comparison function is that we can use have `cmp` use it. Making the unit test use it will be the easier way to guarantee that it works.

Reply:
make sense, updated.

Reply:
let me update BUILD as well...